runtime-components/non-repudiation-app/src/main/scala/com/daml/nonrepudiation/app/Configuration.scala

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.nonrepudiation.app
+
+import java.net.{InetAddress, InetSocketAddress}
+
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+
+object Configuration {
+
+  private val LocalHost = InetAddress.getLocalHost.getHostAddress
+  private val LoopbackAddress = InetAddress.getLoopbackAddress.getHostAddress
+
+  private val ApiDefaultPort: Int = 7882 // Non-repudiation -> NR -> N = 78 and R = 82 in ASCII
+  private val ParticipantDefaultPort: Int = 6865
+  private val ProxyDefaultPort: Int = ParticipantDefaultPort
+
+  val Default: Configuration =
+    Configuration(
+      participantAddress = new InetSocketAddress(LocalHost, ParticipantDefaultPort),
+      proxyAddress = new InetSocketAddress(LoopbackAddress, ProxyDefaultPort),
+      apiAddress = new InetSocketAddress(LoopbackAddress, ApiDefaultPort),
+      apiShutdownTimeout = 10.seconds,
+      databaseJdbcUrl = "jdbc:postgresql:/",
+      databaseJdbcUsername = "nonrepudiation",
+      databaseJdbcPassword = "nonrepudiation",
```

Comment:
I’m slightly sceptical that this is a good idea. Can we instead require users to set those or make them optional and don’t pass any password and user when setting up the hikari thingy? I forgot what the json API and the trigger service do but probably worth trying to keep them in sync.

Reply:
This is already required from the CLI.